packages/gravity/kube-publishing/src/test-runner.ts

```diff
@@ -0,0 +1,163 @@
+      const stat = fs.statSync(filePath);
+      if (stat.isDirectory() && file !== 'node_modules') {
+        await this._modifyApp(filePath, testStep);
+      } else if (stat.isFile()) {
+        // Append to the end of the file.
+        if (file.endsWith('.tsx')) {
+          fs.appendFileSync(filePath, `\n export const test${testStep}=1 \n`, 'utf8');
+        } else if (file.endsWith('.html')) {
+          fs.appendFileSync(filePath, `\n <!-- ${testStep} --> \n`, 'utf8');
+        }
+      }
+    }
+    await sleep(1_000);
+  }
+
+  // Modifyes app build script to generate more test files.
+  private async _modifyBuildStep() {
+    const packageJson = fs.readFileSync(path.join(this._appPath, 'package.json'), 'utf8');
+    // TODO(egorgripasov): Read/preserve original build script to make test more extensible.
+    const newPackageJson = packageJson.replace(
+      /"build": .*/g,
+      `"build": "NODE_OPTIONS=\\"--max-old-space-size=4096\\" tsc --noEmit && vite build && for i in {1..${this._spec.randomFilesCount}}; do base64 /dev/urandom | head -c 5000000 > ./out/test-app/file_$RANDOM$RANDOM_$i.js; done",`
+    );
+    fs.writeFileSync(path.join(this._appPath, 'package.json'), newPackageJson, 'utf8');
+  }
+
+  private async _pubishApp() {
+    // TODO(egorgripasov): Consider using the DX lib directly.
+    await run('npx', ['dx', 'app', 'publish', '--verbose', '--config', path.join(process.cwd(), './config.yml')], {
+      cwd: this._appPath
+    });
+  }
+
+  private async _buildFilesMap(outPath: string, currentPath: string) {
+    const files = fs.readdirSync(currentPath);
+    for await (const file of files) {
+      const filePath = path.join(currentPath, file);
+      const stat = fs.statSync(filePath);
+      if (stat.isDirectory()) {
+        await this._buildFilesMap(outPath, filePath);
+      } else if (stat.isFile()) {
+        const pathDiff = filePath.replace(outPath, '');
+        const appFileContent = fs.readFileSync(filePath, 'utf8');
+        const appFileHash = crypto.createHash('md5').update(appFileContent).digest('hex');
+
+        this._filesMap[pathDiff] = appFileHash;
+      }
+    }
+  }
+
+  private async _evaluateAppFiles(): Promise<EvaluationResult> {
+    const result: EvaluationResult = {};
+    for await (const [pathDiff, expectedHash] of Object.entries(this._filesMap)) {
+      const kubeFilePath = `https://${this._spec.appName}.${this._spec.kubeEndpoint}${pathDiff}`;
+      const response = await fetch(kubeFilePath, { cache: 'no-store' });
+      const kubeFileContent = await response.text();
+      const kubeFileHash = crypto.createHash('md5').update(kubeFileContent).digest('hex');
+
+      const cacheStatus = response.headers.get(cfCacheHeader)
+        ? (response.headers.get(cfCacheHeader) as CacheStatus)
+        : undefined;
+
+      result[kubeFilePath] = {
+        match: expectedHash === kubeFileHash,
+        cacheStatus
+      };
+    }
+
+    return result;
+  }
+
+  async run() {
```

Comment:
in general put the public methods/getters directly below the constructor and private methods below.